lib/src/front_end/chain_builder.dart

```diff
@@ -60,11 +68,40 @@ class ChainBuilder {
   /// The dotted property accesses and method calls following the target.
   final List<ChainCall> _calls = [];
 
-  ChainBuilder(this._visitor, Expression expression) {
-    _unwrapCall(expression);
+  ChainBuilder(this._visitor, this._root) {
+    if (_root case CascadeExpression cascade) {
+      _visitTarget(cascade.target);
+
+      for (var section in cascade.cascadeSections) {
+        _unwrapCall(section);
+      }
+    } else {
+      _unwrapCall(_root);
+    }
   }
 
   Piece build() {
+    if (_root case CascadeExpression cascade) {
+      // If there is only a single section and it can block split, allow it:
```

Comment:
I don't see any code I can correlate to "if there is only a single section". Should this read "If the last section can block split, allow it"?

Reply:
Oops good catch. In practice, this only works if there's only a single section (because cascades are pinned to split if there is more than one), but this code is unnecessarily complex. Fixed.